src/main/java/org/snomed/snowstorm/core/data/services/RelationshipService.java

```diff
@@ -46,9 +52,49 @@
 	@Autowired
 	private ConceptUpdateHelper conceptUpdateHelper;
 
+	@Autowired
+	private MRCMLoader mrcmLoader;
+
 	public Relationship findRelationship(String branchPath, String relationshipId) {
-		Page<Relationship> relationships = findRelationships(branchPath, relationshipId, null, null, null, null, null, null, null, null, PageRequest.of(0, 1));
-		return relationships.getTotalElements() > 0 ? relationships.getContent().get(0) : null;
+		final Page<Relationship> relationships = findRelationships(branchPath, relationshipId, null, null, null, null, null, null, null, null, PageRequest.of(0, 1));
+		if (relationships.isEmpty()) {
+			return null;
+		}
+
+		final Relationship relationship = relationships.getContent().get(0);
+		setConcreteValueFromMRCM(relationship, branchPath);
+
+		return relationship;
+	}
+
+	private void setConcreteValueFromMRCM(final Relationship relationship,
+										  final String branchPath) {
+		final boolean isConcrete = relationship.isConcrete();
+		final boolean isInferred = relationship.getCharacteristicTypeId().equals(Relationship.CharacteristicType.inferred.getConceptId());
+		if (isConcrete && isInferred) {
+			try {
+				final BranchCriteria branchCriteria = versionControlHelper.getBranchCriteria(branchPath);
+				final MRCM mrcm = mrcmLoader.loadActiveMRCM(branchPath, branchCriteria);
+				final List<AttributeRange> attributeRanges = mrcm.getAttributeRanges();
+				final String typeId = relationship.getTypeId();
+				final String value = relationship.getValue();
+
+				for (AttributeRange attributeRange : attributeRanges) {
+					final String referencedComponentId = attributeRange.getReferencedComponentId();
+					if (typeId.equals(referencedComponentId)) {
+						final String rangeConstraint = attributeRange.getRangeConstraint();
+						final String shorthandDataType = rangeConstraint.substring(0, 3);
+
+						relationship.setConcreteValue(
+								ConcreteValue.fromShorthand(value, shorthandDataType)
+						);
```

Comment:
There might be multiple attribute ranges for a given attribute id as the range is 1-to-1 mapping to a attribute domain.

Reply:
I have added a break statement to exit the for-loop.